presto-native-tests/src/test/java/com.facebook.presto.nativetests/TestRepartitionQueriesWithSmallPages.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.nativetests;
+
+import com.facebook.presto.nativeworker.NativeQueryRunnerUtils;
+import com.facebook.presto.nativeworker.PrestoNativeQueryRunnerUtils;
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.AbstractTestRepartitionQueries;
+import com.google.common.collect.ImmutableMap;
+
+public class TestRepartitionQueriesWithSmallPages
+        extends AbstractTestRepartitionQueries
+{
+    private static final String storageFormat = "PARQUET";
+
+    @Override
+    protected QueryRunner createQueryRunner() throws Exception
+    {
+        return PrestoNativeQueryRunnerUtils.createNativeQueryRunner(
```

Comment:
Does native engine support these session properties ? Will native side-car catch these ?

Reply:
Thanks for pointing it out, the property `experimental.optimized-repartitioning` is not applicable to the native engine so removed it. Native engine does support `driver.max-page-partitioning-buffer-size`, which is required for this test, so retained this config.
The native sidecar currently does not throw an error if a session property is set that is not supported by native engine, this is to support [heterogeneous environments](https://github.com/prestodb/presto/pull/23045#pullrequestreview-2355256133). The native sidecar plugin is also not setup on the coordinator currently because of [this limitation](https://github.com/prestodb/presto/issues/24422). Can we enable the native sidecar in these tests subsequently?